l10n_it_delivery_note_inter_warehouse/models/stock_delivery_note.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 Ooops404
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class StockDeliveryNote(models.Model):
+    _inherit = "stock.delivery.note"
+
+    @api.model
+    def _get_sync_fields(self):
+        return [
+            "date",
+            "transport_datetime",
+            "transport_condition_id",
+            "goods_appearance_id",
+            "transport_reason_id",
+            "transport_method_id",
+            "gross_weight",
+            "net_weight",
+            "packages",
+            "volume",
+        ]
+
+    def update_detail_lines(self):
+        res = super().update_detail_lines()
+        for note in self:
+            intercompany_picking_id = note.picking_ids.mapped(
+                "move_lines.inter_warehouse_picking_id"
+            )
+            if intercompany_picking_id:
+                intercompany_note = intercompany_picking_id.delivery_note_id
+                note.partner_ref = intercompany_note.name
+                note.write(
+                    {
+                        field: intercompany_note[field]
+                        for field in self._get_sync_fields()
+                        if intercompany_note[field]
+                    }
+                )
```

Comment:
**issue**: is it possible that `intercompany_picking_id` may contain more than one record?

Reply:
You can definitely have one DDT for multiple pickings, so I don't see why not.

Reply:
@renda-dev perchè c'è un riferimento ai picking intercompany nel modulo interwarehouse?